Build/common.ps1

```diff
@@ -15,12 +15,118 @@ THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
 #>
 
 <#
-.Synopsis
+.DESCRIPTION
+BuildFlags defines operations which the build should perform.
+#>
+[Flags()] enum BuildFlags {
+    Update_PSDSC = 1
+    Update_VSDSC = 2
+    Tests_VSDSC = 4
+    Tests_PSDSC = 8
+    None = 256
+}
+
+<#
+.SYNOPSIS
+Checks if a certain DesiredFlag is included in the InputFlag.
+.DESCRIPTION
+Checks if a certain DesiredFlag is included in the InputFlag.
+The check is performed using a binary and on flags.
+.PARAMETER InputFlag
+Contains all the flags for the current build.
+.PARAMETER DesiredFlag
+The flag that gets checked if it's contained the current build flags.
+#>
+function Test-Flag {
+    [OutputType([bool])]
+    Param (
+        [BuildFlags]
+        $InputFlag,
+
+        [BuildFlags]
+        $DesiredFlag
+    )
+
+    ($InputFlag -band $DesiredFlag) -ne 0
+}
+
+<#
+.SYNOPSIS
+Finds and returns a list of file paths that have been changed since last build
+.DESCRIPTION
+Finds and returns a list of file paths that have been changed since last build.
+The changes are searched in the .git HEAD from the current commit to the last travis commit.
+#>
+function Get-ChangedFiles {
+    [OutputType([System.String[]])]
+    Param()
+
+    $lastTravisCommit = 1
+
+    # find last travis commit
+    while ($true) {
+        $commitInfo = git show HEAD~$lastTravisCommit
+        $author = $commitInfo[1]
+
+        if ($author.Contains('travis@travis-ci.org')) {
+            break
+        }
+
+        $lastTravisCommit += 1
+    }
+
+    # extract names of changed files from HEAD history
+    $changedFiles = git diff --name-only HEAD..HEAD~$lastTravisCommit
+
+    $changedFiles
+}
+<#
+.SYNOPSIS
+Depending on the changes made in the project [BuildFlags] get set and returned.
```

Comment:
Maybe extend the description a little more by describing the possible changes to the project and how that affects the **BuildFlags**.